client/packages.go

```diff
@@ -176,8 +183,9 @@ func (client *Client) Find(opts *FindOptions) ([]*Snap, *ResultInfo, error) {
 	case opts.Private:
 		q.Set("select", "private")
 	}
-	if opts.Section != "" {
-		q.Set("section", opts.Section)
+
+	if section != "" || (section == "" && query == "") {
```

Comment:
Yeah, I don't get that at all. The query being empty is completely irrelevant to whether we send a section or not.